packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -82,7 +91,7 @@ class FlutterPlugin implements Plugin<Project> {
             if (!engineOut.isDirectory()) {
                 throw new GradleException('localEngineOut must point to a local engine build')
```

Comment:
Not directly. `-PlocalEngineOut` (passed to the Gradle build by Flutter Tools) is the full path to the chosen local engine directory, which is a combination of `--local-engine-src-path` and `--local-engine`. If you don't specify `--local-engine-src-path`, it defaults to `$FLUTTER_ROOT/../engine/src`, but the Gradle build script has no idea about that. It just gets the resolved path.

Flutter Tools will have verified the `--local-engine` argument before passing it to Gradle, so you *should* only see this if you manually specify `-PlocalEngineOut` to gradle. Or, you know, Gremlins.